File: frontend/src/app/features/pedidos/domain/interfaces/usecases/IProveedorUseCases.ts
import type { Proveedor } from '../../entities/proveedor';

/**
 * Contrato de los casos de uso de proveedor.
 * @remarks Implementado por {@link ProveedorUseCases}.
 */
export interface IProveedorUseCases {
  /** Devuelve todos los proveedores. */
  getAllAsync(): Promise<Proveedor[]>;
  /** Devuelve sólo los proveedores activos. */
  getActivosAsync(): Promise<Proveedor[]>;
  /**
   * Obtiene un proveedor por su identificador.
   * @param id Identificador del proveedor.
   * @returns Entidad {@link Proveedor} o `null` si no existe.
   */
  getByIdAsync(id: number): Promise<Proveedor | null>;
}

File: backend/ERP.Backend/Domain/Interfaces/IUseCases/IProveedorUseCases.cs
using Application.DTOs;
using Domain.DTOs;
using Domain.Entities;

namespace Application.UseCases
{
    /// <summary>
    /// Interfaz de los casos de uso del maestro de proveedores.
    /// </summary>
    public interface IProveedorUseCase
    {
        /// <summary>
        /// Obtiene todos los proveedores del sistema mapeados a DTO.
        /// </summary>
        /// <returns>Lista de <see cref="ProveedorDTO"/> con todos los registros.</returns>
        Task<List<ProveedorDTO>> GetAllAsync();

        /// <summary>
        /// Obtiene un proveedor por su identificador único mapeado a DTO.
        /// </summary>
        /// <param name="proveedorID">Identificador único del proveedor a buscar.</param>
        /// <returns>El <see cref="ProveedorDTO"/> encontrado, o <c>null</c> si no existe.</returns>
        Task<ProveedorDTO?> GetByIdAsync(int proveedorID);
    }
}